src/prefect/configuration/__init__.py

```diff
@@ -284,45 +277,36 @@ def interpolate_config(config: dict, env_var_prefix: str = None) -> Config:
     return cast(Config, collections.flatdict_to_dict(flat_config, dct_class=Config))
 
 
-def load_configuration(
-    path: str, user_config_path: str = None, env_var_prefix: str = None
-) -> Config:
+def load_configuration(paths: List[str], env_var_prefix: str = None) -> Config:
     """
     Loads a configuration from a known location.
 
     Args:
-        - path (str): the path to the TOML configuration file
-        - user_config_path (str): an optional path to a user config file. If a user config
-            is provided, it will be used to update the main config prior to interpolation
+        - paths (List[str]): a list of paths to TOML configuration files. Each provided path
+            will overwrite matching keys in earlier paths.
         - env_var_prefix (str): any env vars matching this prefix will be used to create
             configuration values
 
     Returns:
         - Config
     """
 
-    # load default config
-    default_config = load_toml(path)
-
-    # load user config
-    if user_config_path and os.path.isfile(str(interpolate_env_vars(user_config_path))):
```

Comment:
@cicdw Do you think this check should be added back for every config option passed in with `paths`?

Reply:
`paths` could contain default, user, custom configs, etc. so either we could error out if a provided config does not exist or we can skip it